lib/nrf_modem_lib/nrf_modem_os.c

```diff
@@ -199,6 +201,15 @@ int32_t nrf_modem_os_timedwait(uint32_t context, int32_t *timeout)
 	return 0;
 }
 
+int nrf_modem_os_sleep(uint32_t timeout)
+{
+	if (timeout == NRF_MODEM_OS_NO_WAIT || timeout == NRF_MODEM_OS_FOREVER) {
+		return -NRF_EINVAL;
+	}
+
+	return k_sleep(K_MSEC(timeout));
+}
```

Comment:
What is the purpose of this function? Does anything use it? Can you add a description?

Reply:
it's part of the OS glue, so it is called by the modem library (like all other functions in this file). The description is in the header file. I have added a description of it in the porting guide as well.